src/OpenConext/Profile/Value/Arp.php

```diff
@@ -142,4 +149,37 @@ public function getAttributesGroupedBySource()
     {
         return $this->arp;
     }
+
+    /**
+     * @return bool
+     */
+    public function hasMotivations()
+    {
+        foreach ($this->arp as $arpSource) {
+            foreach ($arpSource as $arpEntry) {
+                $values = $arpEntry->getValue();
+                $attributeValue = reset($values);
+                if (array_key_exists('motivation', $attributeValue)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public function getMotivationFor(Attribute $attribute)
+    {
+        foreach ($this->arp as $arpSource) {
+            foreach ($arpSource as $arpEntry) {
+                if ($attribute->getAttributeDefinition()->getUrnMace() == $arpEntry->getAttributeDefinition()->getUrnMace()) {
+                    $values = $arpEntry->getValue();
+                    $attributeValue = reset($values);
+                    if (array_key_exists('motivation', $attributeValue)) {
```

Comment:
Do you really want to allow null values here?

I believe isset() in most cases is more idiomatic, array_key_exists() is usually only used when you explicitly want to allow null values.

Reply:
Applies to the two other usages of array_key_exists() to.